core/src/com/group66/game/cannon/GameManager.java

```diff
@@ -0,0 +1,332 @@
+
+    /**
+     * Shoot ball.
+     */
+    public void shootBall() {
+        // TODO add math so ball comes out the top of the cannon?
+        if (canShoot()) {
+            ballManager.shootBall();
+            AudioManager.shoot();
+            timeKeeper.shotTimeReset();
+            this.ballCount++;
+        }
+    }
+
+    /**
+     * Can shoot.
+     *
+     * @return true, if successful
+     */
+    public boolean canShoot() {
+        if (ballManager.getNumberOfShotBalls() > 0
+                || ballManager.getNumberOfPoppingBalls() > 1
+                || ballManager.getNumberOfCannonBalls() == 0) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Gets the roof hitbox.
+     *
+     * @return the roof hitbox
+     */
+    public Rectangle getRoofHitbox() {
+        return this.roofHitbox;
+    }
+
+    /**
+     * Move row down.
+     */
+    public void moveRowDown() {
+        // Move the top hitbox down
+        this.roofHitbox.y -= Config.BALL_DIAM;
+        // Move all the balls down
+        ballManager.moveRowDown();
+        ballGraph.setRoofShift(Config.BALL_DIAM);
+        BustaMove.getGameInstance().log(MessageType.Info, "Moved the roof a row down");
+    }
+
+    /**
+     * Checks if it's game over.
+     *
+     * @return true, if is game over
+     */
+    public boolean isGameOver() {
+        return ballManager.hitsBotom();
+    }
+    
+    /**
+     * Checks if it's game complete.
+     *
+     * @return true, if is game complete
+     */
+    public boolean isGameComplete() {
+        if (ballGraph.numberOfBalls() == 0) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Draw the Balls managed by GameManager.
+     *
+     * @param gameScreen the game screen
+     * @param batch the batch used to draw with
+     * @param delta the delta
+     */
+    public void draw(AbstractGameScreen gameScreen, SpriteBatch batch, float delta) {
+        int xoffset = Config.SINGLE_PLAYER_OFFSET;
+        if (isSplit) {
+            xoffset = Config.SEGMENT_OFFSET * segmentOffset;
+        }
+        
+        //draw the background
+        batch.draw(gameScreen.getBackground(), xoffset + Config.BORDER_SIZE_SIDES, Config.BORDER_SIZE_BOT,
+                Config.LEVEL_WIDTH, Config.LEVEL_HEIGHT);
+        
+        /* Draw the brick wall */
+        batch.draw(gameScreen.getBrickWall(), roofHitbox.x, roofHitbox.y + 10, roofHitbox.width, roofHitbox.height);
+        
+        /* Update the balls */
+        ballManager.update(batch, delta);
+        
+        //draw cannon
+        cannon.draw(batch);
+        
+        /* Draw the score */
+        textDrawer.draw(batch, "Score: " + scoreKeeper.getCurrentScore(), xoffset + Config.SCORE_OFFSET,
+                Config.SCORE_OFFSET);
+    }
+
+    /**
+     * Shift clone.
+     *
+     * @param other the other
+     */
+    public void shiftClone(GameManager other) {
+        //TODO change to iterator
```

Comment:
Would be nice to do this in the future, as now you actually have access to the list. Would be better to just give an iteration IMO. But there was no time to do this now.